### PR TITLE
poc: disable widget resize handlers

### LINKED ISSUE #21027: [Task]: Add a way to specify resize direction for widgets
Some hug widgets should only be resizable in one direction and some others should be resizable in both directions.

Example:
- Image widget should be resizable horizontally & vertically
- Checkbox Group widget should only be resizable horizontally. The height of this widget adjust according to the content.

We need a mechanism to configure this behaviour.

### PR DESCRIPTION
**Pull request by yaldram**
## Description
- Add `disableWidgetResize` config to the autoLayout widget config.
- Disable the resize handlers in vertical and horizontal directions.

> Add a TL;DR when description is extra long (helps content team)

Fixes #21027
> if no issue exists, please create an issue and ask the maintainers about this first

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/components/editorComponents/ResizableComponent.tsx [hunk 1]
```diff
@@ -319,9 +319,11 @@ export const ResizableComponent = memo(function ResizableComponent(
     !isAutoHeightEnabledForWidget(props) ||
     !props.isCanvas;
 
+  // Allow Resize when we select multiple widgets
   const allowResize: boolean =
     !(NonResizableWidgets.includes(props.type) || isMultiSelected) ||
```
[the remark this answers is not in the document]
Comment (yaldram) [reply app/client/src/components/editorComponents/ResizableComponent.tsx:324]: This line of code is used to disable all resize selectors when, we select multiple widgets, so this a total different use case. We should let `NonResizableWidgets` array as it is.

File: app/client/src/widgets/ImageWidget/index.ts [hunk 2]
```diff
@@ -39,6 +39,9 @@ export const CONFIG = {
         },
       },
     ],
+    disableResizeHandles: {
+      horizontal: true,
```
[the remark this answers is not in the document]
Comment (yaldram) [reply app/client/src/widgets/ImageWidget/index.ts:43]: `NonResizbleWidgets` is a separate array used for disabling all resize handlers when multiple widgets are selected, it is not used to disable resize handlers for widgets laying on the canvas.

File: app/client/src/components/editorComponents/ResizableUtils.ts [hunk 3]
```diff
@@ -99,9 +99,33 @@ export function isHandleResizeAllowed(
     direction === ReflowDirection.LEFT ||
     direction === ReflowDirection.RIGHT
   ) {
+    // remove all the horizontal resize handlers for fill widgets
     return isFlexChild && responsiveBehavior === ResponsiveBehavior.Fill
```
Comment (aswathkk) [inline app/client/src/components/editorComponents/ResizableUtils.ts:103]: I feel like this condition could be moved into `isResizingDisabled`.
That way `isHandleResizeAllowed` will only deals with fixed mode resize

Comment (aswathkk) [reply app/client/src/components/editorComponents/ResizableUtils.ts:103]: Needs @marks0351's input here

**Review by aswathkk — COMMENTED** [review]
Added minor comments.
I will add @marks0351 to review changes related to core resizing code.

File: app/client/src/components/editorComponents/ResizableUtils.ts (hunk 3 above)
Comment (marks0351) [reply app/client/src/components/editorComponents/ResizableUtils.ts:103]: @yaldram Sry for bringing this up late, cud u cut the mobile/v1/qa branch, coz i have actually restructured resizing and have separated auto layout resizing. so u cud pick these changes and apply to the auto layout's resize only.

**Review by marks0351 — CHANGES_REQUESTED** [review]

File: app/client/src/resizable/autolayoutresize/index.tsx [hunk 4]
```diff
@@ -572,6 +576,16 @@ export function ReflowResizable(props: ResizableProps) {
       props.isFlexChild,
       props.responsiveBehavior,
     );
+
+    const { disableResizeHandles } = WidgetFactory.getWidgetAutoLayoutConfig(
+      widget.type,
+    );
+
+    const disableResizing = isResizingDisabled(
```
Comment (aswathkk) [inline app/client/src/resizable/autolayoutresize/index.tsx:584]: @yaldram Do you think it is better to move the logic of disabling of horizontal resize for fill widgets here?
If we do so, `isHandleResizeAllowed` will used only for fixed mode resize.

File: app/client/src/widgets/InputWidgetV2/index.ts [hunk 5]
```diff
@@ -41,7 +41,7 @@ export const CONFIG = {
   },
   autoLayout: {
     defaults: {
-      rows: 6.6,
```
Comment (aswathkk) [inline app/client/src/widgets/InputWidgetV2/index.ts:44]: great job in catching this and fixing it

File: app/client/src/components/editorComponents/ResizableUtils.ts [hunk 6]
```diff
@@ -90,18 +90,44 @@ export function isHandleResizeAllowed(
...
   if (direction === ReflowDirection.TOP || direction === ReflowDirection.BOTTOM)
     return verticalEnabled;
   else if (
     direction === ReflowDirection.LEFT ||
     direction === ReflowDirection.RIGHT
   ) {
-    return isFlexChild && responsiveBehavior === ResponsiveBehavior.Fill
-      ? false
-      : horizontalEnabled;
+    // remove all the horizontal resize handlers for fill widgets
```
Comment (aswathkk) [inline app/client/src/components/editorComponents/ResizableUtils.ts:100]: This comment is irrelevant

**Review by aswathkk — CHANGES_REQUESTED** [review]
@yaldram 
A comment needs to be cleaned up.
Also, can you resolve the merge conflict?

**Review by aswathkk — APPROVED** [review]

**Pull request merged**